modules/geo/src/main/java/org/opensearch/geo/search/aggregations/bucket/geogrid/cells/GeoShapeCellValues.java

```diff
@@ -67,7 +67,7 @@ public BoundedCellValues(
             final GeoBoundingBox boundingBox
         ) {
             super(geoShapeValue, precision, encoder);
-            this.geoBoundingBox = boundingBox;
+            this.geoBoundingBox = new Rectangle(boundingBox.left(), boundingBox.right(), boundingBox.top(), boundingBox.bottom());
```

Comment:
Why create a new Rectangle instance here? Was there a concurrency issue?

Reply:
I am converting a GeoBoundingBox instance to a rectangle geometry instance as so that I can figure out if the indexed shapes are intersecting with the GeoBoundingBox.

Actually the GeoBoundingBox class doesn't have a method to figure out the intersection with geometry hence I had to convert it first into rectangle instance and then find the intersection.


There is no concurrency issue. This rectangle instance will be created once.